btf/core.go

```diff
@@ -169,19 +169,31 @@ func (k coreKind) String() string {
 //
 // Fixups are returned in the order of relos, e.g. fixup[i] is the solution
 // for relos[i].
-func CORERelocate(relos []*CORERelocation, target *Spec, bo binary.ByteOrder, resolveLocalTypeID func(Type) (TypeID, error)) ([]COREFixup, error) {
+func CORERelocate(relos []*CORERelocation, target *Spec, kmodTarget *Spec, kmodName string, bo binary.ByteOrder, resolveLocalTypeID func(Type) (TypeID, error)) ([]COREFixup, error) {
 	if target == nil {
 		var err error
 		target, _, err = kernelSpec()
 		if err != nil {
 			return nil, fmt.Errorf("load kernel spec: %w", err)
 		}
 	}
-
 	if bo != target.byteOrder {
 		return nil, fmt.Errorf("can't relocate %s against %s", bo, target.byteOrder)
 	}
 
+	if kmodTarget == nil && kmodName != "" {
```

Comment:
This is to avoid copies of the BTF I imagine? Otherwise `kmodName` seems unused.

Reply:
`kmodTarget` can `nil` in both the non-kmod function and kmod function cases if they are expecting to use default BTF. `kmodName` is used to disambiguate between those cases.

Reply:
`kmodName` is also passed to `kernelModuleSpec`